clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -986,22 +987,33 @@ private TxnRequestHandler addPartitionsToTransactionHandler() {
 
     private TxnOffsetCommitHandler txnOffsetCommitHandler(TransactionalRequestResult result,
                                                           Map<TopicPartition, OffsetAndMetadata> offsets,
-                                                          String consumerGroupId) {
+                                                          ConsumerGroupMetadata groupMetadata,
+                                                          boolean enableGroupFencing) {
```

Comment:
nit: I think we can do away with `enableGroupFencing` and derive its value from `ConsumerGroupMetadata`. In spite of my comment on the previous PR, it may be simpler to just let the defaults be consistent with the expected default values and just have one path below.

Reply:
I think we should validate the object when it is received in `sendOffsets`.